dev-tools/packaging/package_test.go

```diff
@@ -118,14 +117,15 @@ func checkDeb(t *testing.T, file string, buf *bytes.Buffer) {
 		return
 	}
 
+	// deb file permissions are managed post-install
```

Comment:
Interesting.

Reply:
Perhaps that should read `deb file permission can be managed post-install`, as detailed in https://github.com/elastic/beats/pull/8450#discussion_r221070731.  I am much more familiar with rpm packaging than deb but after reviewing the docs, etc I think it's a reasonable approach.

Reply:
That comment got me wondering, did you try modifying these lines to set the configured user/group?

https://github.com/elastic/beats/blob/1bfe5bb6c11d37818ec5d4016c7f0996fe0bf8e2/dev-tools/mage/pkgtypes.go#L772-L773

Reply:
Good idea, I had not.  Results:

That did change the ownership within the tar ball used as input to `fpm` but initially made no change in the rpm or deb (rpms still using `%attr`, deb used root).  With `--deb-use-file-permissions` (which I misspoke about earlier, is not on by default) permissions came through as those from `os.Getuid()` and `os.Getgid()`.  If we could know the uid/gid on the target system ahead of time this would work, otherwise I don't see a path forward with this approach.

As before, I'm not sure it's that important since there doesn't seem to be a reset uids/gids deb equivalent to rpm's `setugids`, other than that it makes post-install crucial, but I'm interested to hear other thoughts on this.

Reply:
FPM must be unpacking the tar in someway that it loses ownership information along the way. It could be because we execute fpm [as a non-root user](https://github.com/elastic/beats/blob/1bfe5bb6c11d37818ec5d4016c7f0996fe0bf8e2/dev-tools/mage/pkgtypes.go#L619-L622) so then when it unpacks the tar the files can only be owned by that non-root user b/c tar cannot chown them. If FPM were to read the attributes directly from the tar then I would expect this to work.

You could try commenting out the lines that cause FPM to be executed as non-root and see if that changes anything. If not, I'm good with what you have.

Reply:
That did change the result as you anticipated.  Unfortunately the uid/gid came through rather than the uname/gname, meaning that unless we know the beat user's uid/gid ahead of time, which we can't, we're back to chowning post-install, which we should do from root-owned to beat-user-owned.